[shiftmedia-libgnutls] build failure

### Operating system

windows

### Compiler

new msvc (not sure the version)

### Steps to reproduce the behavior

```Shell
vcpkg install shiftmedia-libgnutls
```


### Failure logs

Failure logs report by @MonicaLiu0311 
```
PS E:\ngtcp2> ./vcpkg install ngtcp2[*]
Computing installation plan...
The following packages will be built and installed:
    ngtcp2[core,gnutls,libressl,wolfssl]:x64-windows@1.7.0
  * shiftmedia-libgnutls:x64-windows@3.8.4#2
  * wolfssl[core,quic]:x64-windows@5.7.2#2
Additional packages (*) will be modified to complete this operation.
Detecting compiler hash for triplet x64-windows...
Compiler found: C:/Program Files/Microsoft Visual Studio/2022/Enterprise/VC/Tools/MSVC/14.41.34120/bin/Hostx64/x64/cl.exe
Restored 0 package(s) from C:\Users\monica\AppData\Local\vcpkg\archives in 168 us. Use --debug to see more details.
Installing 1/3 shiftmedia-libgnutls:x64-windows@3.8.4#2...
Building shiftmedia-libgnutls:x64-windows@3.8.4#2...
-- Using cached ShiftMediaProject-gnutls-3.8.4.tar.gz.
-- Cleaning sources at E:/ngtcp2/buildtrees/shiftmedia-libgnutls/src/3.8.4-2d4ec1f875.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source E:/ngtcp2/downloads/ShiftMediaProject-gnutls-3.8.4.tar.gz
-- Applying patch external-libtasn1.patch
-- Applying patch pkgconfig.patch
-- Applying patch ssize_t_already_define.patch
-- Applying patch fix-warnings.patch
-- Using source at E:/ngtcp2/buildtrees/shiftmedia-libgnutls/src/3.8.4-2d4ec1f875.clean
-- Downloading https://git.savannah.gnu.org/gitweb/?p=gnulib.git\;a=snapshot\;h=3639c57\;sf=tgz -> gnulib-3639c57.tar.gz...
-- Extracting source E:/ngtcp2/downloads/gnulib-3639c57.tar.gz
-- Using source at E:/ngtcp2/buildtrees/shiftmedia-libgnutls/src/3639c57-6c17dd416b.clean
-- Building SMP/libgnutls.sln for Release
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:127 (message):
    Command failed: msbuild E:/ngtcp2/buildtrees/shiftmedia-libgnutls/x64-windows-rel/3.8.4-2d4ec1f875.clean/SMP/libgnutls.sln /p:Configuration=ReleaseDLL /p:YasmPath="E:\\ngtcp2\\installed\\x64-windows\\tools\\yasm\\yasm.exe" /p:OutDir=..\\msvc /t:Rebuild /p:Platform=x64 /p:PlatformToolset=v143 /p:VCPkgLocalAppDataDisabled=true /p:UseIntelMKL=No /p:WindowsTargetPlatformVersion=10.0.22621.0 /p:VcpkgTriplet=x64-windows /p:VcpkgInstalledDir=E:/ngtcp2/installed /p:VcpkgManifestInstall=false /p:UseMultiToolTask=true /p:MultiProcMaxCount=9 /p:EnforceProcessCountAcrossBuilds=true /m:9 /p:ForceImportBeforeCppTargets=E:/ngtcp2/scripts/buildsystems/msbuild/vcpkg.targets /p:VcpkgApplocalDeps=false /p:RuntimeLibrary=MultiThreadedDLL
    Working Directory: E:/ngtcp2/buildtrees/shiftmedia-libgnutls/x64-windows-rel/3.8.4-2d4ec1f875.clean
    Error code: 1
    See logs for more information:
      E:\ngtcp2\buildtrees\shiftmedia-libgnutls\build-x64-windows-rel-out.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_install_msbuild.cmake:79 (vcpkg_execute_required_process)
  ports/shiftmedia-libgnutls/portfile.cmake:137 (vcpkg_install_msbuild)
  scripts/ports.cmake:192 (include)
```

```
     2>E:\ngtcp2\buildtrees\shiftmedia-libgnutls\x64-windows-rel\3.8.4-2d4ec1f875.clean\lib\verify-tofu.c(496,21): error C2197: 'int mkdir(const char *)': too many arguments for call [E:\ngtcp2\buildtrees\shiftmedia-libgnutls\x64-windows-rel\3.8.4-2d4ec1f875.clean\SMP\libgnutls.vcxproj]
     2>E:\ngtcp2\buildtrees\shiftmedia-libgnutls\x64-windows-rel\3.8.4-2d4ec1f875.clean\lib\verify-tofu.c(597,21): error C2197: 'int mkdir(const char *)': too many arguments for call [E:\ngtcp2\buildtrees\shiftmedia-libgnutls\x64-windows-rel\3.8.4-2d4ec1f875.clean\SMP\libgnutls.vcxproj]
         C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.41.34120\bin\HostX64\x64\CL.exe /c /I.\ /I.\lib /I..\lib /I..\lib\accelerated /I..\lib\auth /I..\lib\ext /I..\lib\extras /I..\lib\minitasn1 /I..\lib\nettle /I..\lib\nettle\int /I..\lib\x509 /I.\lib\x509 /I..\lib\includes /I..\lib\accelerated\x86 /I..\gnulib\lib /I.\gnulib\lib /I.\gnulib\lib\uninorm /I.\libtasn1 /I..\devel\libtasn1\lib /I..\msvc\\include /I"E:\ngtcp2\buildtrees\shiftmedia-libgnutls\x64-windows-rel\3.8.4-2d4ec1f875.clean\SMP\\..\..\prebuilt\include" /I"E:/ngtcp2/installed\x64-windows\include" /Zi /nologo /W3 /WX- /diagnostics:column /FS /O2 /Ob2 /Oi /Ot /Oy /GT /GL /D HAVE_CONFIG_H /D inline=__inline /D ASN1_BUILDING /D ASN1_STATIC /D "LOCALEDIR=\".\"" /D __func__=__FUNCTION__ /D ASM_X86 /D USE_WINDOWS_THREADS /D "GNUTLS_LIBRARY_SONAME=\"gnutls\"" /D "NETTLE_LIBRARY_SONAME=\"nettle\"" /D "HOGWEED_LIBRARY_SONAME=\"hogweed\"" /D "GMP_LIBRARY_SONAME=\"gmp\"" /D ASM_X86_64 /D _WINDOWS /D WIN32 /D _WIN32_WINNT=0x0601 /D NDEBUG /D _USRDLL /D _CRT_SECURE_NO_DEPRECATE /D _CRT_NONSTDC_NO_DEPRECATE /D _CRT_SECURE_NO_WARNINGS /D _WINDLL /D _UNICODE /D UNICODE /GF /Gm- /EHsc /MD /GS /Gy /fp:precise /Zc:wchar_t /Zc:forScope /Zc:inline /Fo"E:\ngtcp2\buildtrees\shiftmedia-libgnutls\x64-windows-rel\3.8.4-2d4ec1f875.clean\SMP\obj\ReleaseDLL\x64\libgnutls\\x509_extensions.obj" /Fd"E:\ngtcp2\buildtrees\shiftmedia-libgnutls\x64-windows-rel\3.8.4-2d4ec1f875.clean\SMP\obj\ReleaseDLL\x64\libgnutls\gnutls.pdb" /external:W3 /Gd /TC /wd4267 /wd4244 /wd4018 /wd4090 /wd4047 /wd4024 /wd4311 /wd4116 /wd4146 /wd4312 /wd4028 /wd4334 /wd4101 /wd4013 /wd4020 /wd4068 /wd4003 /FC /errorReport:queue /we4113 ..\lib\x509\extensions.c
         C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.41.34120\bin\HostX64\x64\CL.exe /c /I.\ /I.\lib /I..\lib /I..\lib\accelerated /I..\lib\auth /I..\lib\ext /I..\lib\extras /I..\lib\minitasn1 /I..\lib\nettle /I..\lib\nettle\int /I..\lib\x509 /I.\lib\x509 /I..\lib\includes /I..\lib\accelerated\x86 /I..\gnulib\lib /I.\gnulib\lib /I.\gnulib\lib\uninorm /I.\libtasn1 /I..\devel\libtasn1\lib /I..\msvc\\include /I"E:\ngtcp2\buildtrees\shiftmedia-libgnutls\x64-windows-rel\3.8.4-2d4ec1f875.clean\SMP\\..\..\prebuilt\include" /I"E:/ngtcp2/installed\x64-windows\include" /Zi /nologo /W3 /WX- /diagnostics:column /FS /O2 /Ob2 /Oi /Ot /Oy /GT /GL /D HAVE_CONFIG_H /D inline=__inline /D ASN1_BUILDING /D ASN1_STATIC /D "LOCALEDIR=\".\"" /D __func__=__FUNCTION__ /D ASM_X86 /D USE_WINDOWS_THREADS /D "GNUTLS_LIBRARY_SONAME=\"gnutls\"" /D "NETTLE_LIBRARY_SONAME=\"nettle\"" /D "HOGWEED_LIBRARY_SONAME=\"hogweed\"" /D "GMP_LIBRARY_SONAME=\"gmp\"" /D ASM_X86_64 /D _WINDOWS /D WIN32 /D _WIN32_WINNT=0x0601 /D NDEBUG /D _USRDLL /D _CRT_SECURE_NO_DEPRECATE /D _CRT_NONSTDC_NO_DEPRECATE /D _CRT_SECURE_NO_WARNINGS /D _WINDLL /D _UNICODE /D UNICODE /GF /Gm- /EHsc /MD /GS /Gy /fp:precise /Zc:wchar_t /Zc:forScope /Zc:inline /Fo"E:\ngtcp2\buildtrees\shiftmedia-libgnutls\x64-windows-rel\3.8.4-2d4ec1f875.clean\SMP\obj\ReleaseDLL\x64\libgnutls\\" /Fd"E:\ngtcp2\buildtrees\shiftmedia-libgnutls\x64-windows-rel\3.8.4-2d4ec1f875.clean\SMP\obj\ReleaseDLL\x64\libgnutls\gnutls.pdb" /external:W3 /Gd /TC /wd4267 /wd4244 /wd4018 /wd4090 /wd4047 /wd4024 /wd4311 /wd4116 /wd4146 /wd4312 /wd4028 /wd4334 /wd4101 /wd4013 /wd4020 /wd4068 /wd4003 /FC /errorReport:queue /we4113 "..\lib\x509\hostname-verify.c"
         C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.41.34120\bin\HostX64\x64\CL.exe /c /I.\ /I.\lib /I..\lib /I..\lib\accelerated /I..\lib\auth /I..\lib\ext /I..\lib\extras /I..\lib\minitasn1 /I..\lib\nettle /I..\lib\nettle\int /I..\lib\x509 /I.\lib\x509 /I..\lib\includes /I..\lib\accelerated\x86 /I..\gnulib\lib /I.\gnulib\lib /I.\gnulib\lib\uninorm /I.\libtasn1 /I..\devel\libtasn1\lib /I..\msvc\\include /I"E:\ngtcp2\buildtrees\shiftmedia-libgnutls\x64-windows-rel\3.8.4-2d4ec1f875.clean\SMP\\..\..\prebuilt\include" /I"E:/ngtcp2/installed\x64-windows\include" /Zi /nologo /W3 /WX- /diagnostics:column /FS /O2 /Ob2 /Oi /Ot /Oy /GT /GL /D HAVE_CONFIG_H /D inline=__inline /D ASN1_BUILDING /D ASN1_STATIC /D "LOCALEDIR=\".\"" /D __func__=__FUNCTION__ /D ASM_X86 /D USE_WINDOWS_THREADS /D "GNUTLS_LIBRARY_SONAME=\"gnutls\"" /D "NETTLE_LIBRARY_SONAME=\"nettle\"" /D "HOGWEED_LIBRARY_SONAME=\"hogweed\"" /D "GMP_LIBRARY_SONAME=\"gmp\"" /D ASM_X86_64 /D _WINDOWS /D WIN32 /D _WIN32_WINNT=0x0601 /D NDEBUG /D _USRDLL /D _CRT_SECURE_NO_DEPRECATE /D _CRT_NONSTDC_NO_DEPRECATE /D _CRT_SECURE_NO_WARNINGS /D _WINDLL /D _UNICODE /D UNICODE /GF /Gm- /EHsc /MD /GS /Gy /fp:precise /Zc:wchar_t /Zc:forScope /Zc:inline /Fo"E:\ngtcp2\buildtrees\shiftmedia-libgnutls\x64-windows-rel\3.8.4-2d4ec1f875.clean\SMP\obj\ReleaseDLL\x64\libgnutls\\" /Fd"E:\ngtcp2\buildtrees\shiftmedia-libgnutls\x64-windows-rel\3.8.4-2d4ec1f875.clean\SMP\obj\ReleaseDLL\x64\libgnutls\gnutls.pdb" /external:W3 /Gd /TC /wd4267 /wd4244 /wd4018 /wd4090 /wd4047 /wd4024 /wd4311 /wd4116 /wd4146 /wd4312 /wd4028 /wd4334 /wd4101 /wd4013 /wd4020 /wd4068 /wd4003 /FC /errorReport:queue /we4113 "..\lib\x509\ip.c"
         extensions.c
         hostname-verify.c
         ip.c
     2>Done Building Project "E:\ngtcp2\buildtrees\shiftmedia-libgnutls\x64-windows-rel\3.8.4-2d4ec1f875.clean\SMP\libgnutls.vcxproj" (Rebuild target(s)) -- FAILED.
     1>Done Building Project "E:\ngtcp2\buildtrees\shiftmedia-libgnutls\x64-windows-rel\3.8.4-2d4ec1f875.clean\SMP\libgnutls.sln" (Rebuild target(s)) -- FAILED.

Build FAILED.
```


### Additional context

I cannot reproduce the problem. I have old msvc. 

```cmd
C:\Program Files\Microsoft Visual Studio\2022\Community>msbuild --version
MSBuild version 17.5.1+f6fdcf537 for .NET Framework
17.5.1.16304
C:\Program Files\Microsoft Visual Studio\2022\Community>cl
Microsoft (R) C/C++ Optimizing Compiler Version 19.35.32216.1 for x64
Copyright (C) Microsoft Corporation.  All rights reserved.

usage: cl [ option... ] filename... [ /link linkoption... ]

C:\Program Files\Microsoft Visual Studio\2022\Community>
```

@MonicaLiu0311 Can you bring more details on your msvc version / logs / other detail to this issue?

--- a/ports/shiftmedia-libgnutls/mkdir.patch
+++ b/ports/shiftmedia-libgnutls/mkdir.patch
@@ -1,0 +1,13 @@
+diff --git a/SMP/unistd.h b/SMP/unistd.h
+index b54d4be15..273204e00 100644
+--- a/SMP/unistd.h
++++ b/SMP/unistd.h
+@@ -32,6 +32,8 @@
+ #include <direct.h>
+ #include <fcntl.h>
+ 
++#define mkdir(path, mode) _mkdir(path)
++
+ #define R_OK    4       /* Test for read permission.  */
+ #define W_OK    2       /* Test for write permission.  */
+ //#define   X_OK    1       /* execute permission - unsupported in windows*/

--- a/ports/shiftmedia-libgnutls/portfile.cmake
+++ b/ports/shiftmedia-libgnutls/portfile.cmake
@@ -11,6 +11,7 @@ vcpkg_from_github(
         pkgconfig.patch
         ssize_t_already_define.patch
         fix-warnings.patch
+        mkdir.patch
 )
 
 file(REMOVE_RECURSE "${SOURCE_PATH}/devel/perlasm")

--- a/ports/shiftmedia-libgnutls/vcpkg.json
+++ b/ports/shiftmedia-libgnutls/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "shiftmedia-libgnutls",
   "version": "3.8.4",
-  "port-version": 2,
+  "port-version": 3,
   "description": "Unofficial GnuTLS fork with added custom native Visual Studio project build tools. ",
   "homepage": "https://github.com/ShiftMediaProject/gnutls",
   "license": "LGPL-2.1-only",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -8214,7 +8214,7 @@
     },
     "shiftmedia-libgnutls": {
       "baseline": "3.8.4",
-      "port-version": 2
+      "port-version": 3
     },
     "shiftmedia-libgpg-error": {
       "baseline": "1.45",

--- a/versions/s-/shiftmedia-libgnutls.json
+++ b/versions/s-/shiftmedia-libgnutls.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "b18bd4e582995b852d1b9eac4b79805d84aaa656",
+      "version": "3.8.4",
+      "port-version": 3
+    },
+    {
       "git-tree": "1d4b38dbf344dc537c0f0e8fae77706d09ef0126",
       "version": "3.8.4",
       "port-version": 2

